tests/Aspire.Hosting.Tests/ResourceExtensionsTests.cs

```diff
@@ -25,4 +25,102 @@ public void TryGetContainerImageNameReturnsCorrectFormatWhenShaNotSupplied()
+                Assert.Equal("single-node", env.Value);
+            },
+            env =>
+            {
+                Assert.Equal("xpack.security.enabled", env.Key);
+                Assert.Equal("true", env.Value);
+            },
+            env =>
+            {
+                Assert.Equal("ELASTIC_PASSWORD", env.Key);
+                Assert.False(string.IsNullOrEmpty(env.Value));
+            });
+    }
+
+    [Fact]
+    public async Task GetEnvironmentVariableValuesAsyncReturnCorrectVariablesUsingValueProviderInRunMode()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+        builder.Configuration["Parameters:ElasticPassword"] = "123456";
+
+        var passwordParameter = builder.AddParameter("ElasticPassword");
+
+        var container = builder.AddContainer("elasticsearch", "library/elasticsearch", "8.14.0")
+         .WithEnvironment("discovery.type", "single-node")
+         .WithEnvironment("xpack.security.enabled", "true")
+         .WithEnvironment("ELASTIC_PASSWORD", passwordParameter);
+
+        var env = await container.Resource.GetEnvironmentVariableValuesAsync();
+
+        Assert.Collection(env,
+            env =>
+            {
+                Assert.Equal("discovery.type", env.Key);
+                Assert.Equal("single-node", env.Value);
+            },
+            env =>
+            {
+                Assert.Equal("xpack.security.enabled", env.Key);
+                Assert.Equal("true", env.Value);
+            },
+            env =>
+            {
+                Assert.Equal("ELASTIC_PASSWORD", env.Key);
+                Assert.False(string.IsNullOrEmpty(env.Value));
+            });
+    }
+
+    [Fact]
+    public async Task GetEnvironmentVariableValuesAsyncReturnCorrectVariablesUsingManifestExpressionProviderInPublishMode()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+        builder.Configuration["Parameters:ElasticPassword"] = "123456";
+
+        var passwordParameter = builder.AddParameter("ElasticPassword");
+
+        var container = builder.AddContainer("elasticsearch", "library/elasticsearch", "8.14.0")
+         .WithEnvironment("discovery.type", "single-node")
+         .WithEnvironment("xpack.security.enabled", "true")
+         .WithEnvironment("ELASTIC_PASSWORD", passwordParameter);
+
+        var env = await container.Resource.GetEnvironmentVariableValuesAsync(DistributedApplicationOperation.Publish);
+
+        Assert.Collection(env,
+            env =>
+            {
+                Assert.Equal("discovery.type", env.Key);
+                Assert.Equal("single-node", env.Value);
+            },
+            env =>
+            {
+                Assert.Equal("xpack.security.enabled", env.Key);
+                Assert.Equal("true", env.Value);
+            },
+            env =>
+            {
+                Assert.Equal("{ElasticPassword.value}", env.Value);
+                Assert.False(string.IsNullOrEmpty(env.Value));
```

Comment:
We should verify the Value is the correct parameter reference in publish mode, not just that it isn't null or empty.

Reply:
Reactivating. This hasn't been resolved yet.